src/Compilers/CSharp/Portable/Symbols/ParameterSymbol.cs

```diff
@@ -385,6 +385,8 @@ internal sealed override ObsoleteAttributeData ObsoleteAttributeData
 
         internal abstract FlowAnalysisAnnotations FlowAnalysisAnnotations { get; }
 
+        internal abstract bool HasEnumeratorCancellationAttribute { get; }
```

Comment:
If we never need to know this except for source parameters, why is this property on all parameters?

Reply:
I've taken the inheritance approach, rather than type check. I can take a stab and we can decide.

Reply:
Done